bpo-46483: remove `__class_getitem__` from `pathlib.PurePath` (#30848)
python/cpython: Misc/NEWS.d/next/Library/2022-01-24-13-00-09.bpo-46483.9XnmKp.rst

> Remove ``__class_getitem__`` from :class:`pathlib.PurePath`

Reviewer: It might be good to:

1. Give a one-sentence description of why it's being removed. 
2. Delete the news entry that was added in #30822 (it'll make for a pretty confusing changelog otherwise).

> Remove ``__class_getitem__`` from :class:`pathlib.PurePath`

Reviewer 2: ```suggestion
Remove ``__class_getitem__`` from :class:`pathlib.PurePath`.
```

> Remove ``__class_getitem__`` from :class:`pathlib.PurePath`.
> 
> Why? Because this class was not really generic:
> ``PurePath`` only works with ``str``.

Reviewer 3: ```suggestion
Remove :meth:`~object.__class_getitem__` from :class:`pathlib.PurePath` as this class was not supposed to be generic.
```
